ui/pages/permissions-connect/connect-page/utils.ts

```diff
@@ -0,0 +1,83 @@
+import { Hex } from '@metamask/utils';
+import {
+  Caip25CaveatType,
+  Caip25CaveatValue,
+  Caip25EndowmentPermissionName,
+  setEthAccounts,
+  setPermittedEthChainIds,
+} from '@metamask/multichain';
+
+export type PermissionsRequest = Record<
+  string,
+  { caveats?: { type: string; value: Caip25CaveatValue }[] }
+>;
+
+/**
+ * Takes in an incoming {@link PermissionsRequest} and attempts to return the {@link Caip25CaveatValue} with the Ethereum accounts set.
+ *
+ * @param permissions - The {@link PermissionsRequest} with the target name of the {@link Caip25EndowmentPermissionName}.
+ * @returns The {@link Caip25CaveatValue} with the Ethereum accounts set.
+ */
+export function getRequestedSessionScopes(
+  permissions?: PermissionsRequest,
+): Pick<Caip25CaveatValue, 'requiredScopes' | 'optionalScopes'> {
+  return (
+    permissions?.[Caip25EndowmentPermissionName]?.caveats?.find(
+      (caveat) => caveat.type === Caip25CaveatType,
+    )?.value ?? {
+      optionalScopes: {},
+      requiredScopes: {},
+    }
+  );
+}
+
+/**
+ * Parses the CAIP-25 authorized permissions object after UI confirmation.
+ *
+ * @param addresses - The list of permitted addresses.
+ * @param hexChainIds - The list of permitted chains.
+ * @returns The granted permissions with the target name of the {@link Caip25EndowmentPermissionName}.
+ */
+export function parseCaip25PermissionsResponse(
```

Comment:
[nit] - I think `parse` maybe is a confusing word for this. 

maybe just keep it pretty generic/straightforward:
```suggestion
export function getApprovedSessionsScopes(
```

Reply:
resolved in [ecad6ee](https://github.com/MetaMask/metamask-extension/commit/ecad6eee200b9d39a338e11f78d2588969223fba)